### PR TITLE
[Feature]: Track my preference for ranked vs unranked games, resolves…

### PR DESCRIPTION
**Pull request by seriouslysean**
… #109

## Please check the following

- [x] Do the tests still pass? (see [Run the Tests](https://github.com/cuttle-cards/cuttle#run-the-tests))
- [x] Is the code formatted properly? (see [Linting (Formatting)](https://github.com/cuttle-cards/cuttle#linting-formatting))
- For New Features:
  - [x] Have tests been added to cover any new features or fixes?
  - [x] Has the documentation been updated accordingly?

### CODE REVIEW

File: client/js/components/CreateGameDialog.vue [hunk 1]
```diff
@@ -87,6 +97,17 @@ export default {
...
+      try {
+        localStorage.setItem(LS_PREFERS_RANKED_NAME, prefersRanked === true);
+      } catch (err) {
+        // Local storage is full or something else happened so don't set the value
+      }
+      this.isRanked = prefersRanked;
+    },
+    getRankedPreference() {
+      return localStorage.getItem(LS_PREFERS_RANKED_NAME) === 'true';
+    },
```
Comment (itsalaidbacklife) [inline client/js/components/CreateGameDialog.vue:110]: Woo venturing into new lands here. It begs questions around future features and when to store data in the db vs local storage etc but I don't think we need to establish a comprehensive plan for that now.

File: client/js/components/CreateGameDialog.vue [hunk 2]
```diff
@@ -63,6 +65,14 @@ export default {
...
     };
   },
+  watch: {
+    isRanked(isRanked) {
+      this.setRankedPreference(isRanked);
+    },
+  },
+  mounted() {
+    this.isRanked = this.getRankedPreference();
+  },
```
Comment (itsalaidbacklife) [inline client/js/components/CreateGameDialog.vue:75]: Any reason to prefer `mounted()` over `created()`?
```suggestion
  created() {
    this.isRanked = this.getRankedPreference();
  },
```

File: tests/e2e/specs/out-of-game/home.spec.js [hunk 3]
```diff
@@ -150,6 +150,37 @@ describe('Home - Game List', () => {
 
 describe('Home - Create Game', () => {
   beforeEach(setup);
+
+  it.only('Saves ranked setting between sessions', () => {
```
Comment (itsalaidbacklife) [inline tests/e2e/specs/out-of-game/home.spec.js:154]:
```suggestion
  it('Saves ranked setting between sessions', () => {
```

**Review by itsalaidbacklife — APPROVED** [review]
Clean implementation. The second watcher in the app and the first use of localStorage! Exciting stuff

File: client/js/components/CreateGameDialog.vue (hunk 2 above)
Comment (seriouslysean) [reply client/js/components/CreateGameDialog.vue:75]: I don't think we should do this. `mounted()` is just as good and while we're not a SSR app, created has the window only in the browser (which would require an `if (window)` check) or something with `process.env.VUE_ENV === 'client'` if we ever looked in to it. We get the same benefit with mounted and the knowledge that window is always available within it.

File: client/js/components/CreateGameDialog.vue (hunk 1 above)
Comment (seriouslysean) [reply client/js/components/CreateGameDialog.vue:110]: We should create `localStorage` utilities once we start using this in more places, but ideally we'll add them to a user profile object. Maybe LS as a fallback for guests if we ever go that route.

File: tests/e2e/specs/out-of-game/home.spec.js (hunk 3 above)
Comment (seriouslysean) [reply tests/e2e/specs/out-of-game/home.spec.js:154]: Whoops, thank you. Missed that 😓 -- really need a precommit hook to catch those, haha.

**Pull request merged**